server/plugin/configuration.go

```diff
@@ -26,7 +26,7 @@ type Configuration struct {
 	EncryptionKey           string
 	EnterpriseBaseURL       string
 	EnterpriseUploadURL     string
-	EnableCodePreview       bool
```

Comment:
@larkox How would an admin select one options from the new setting? By default, the first element of a dropdown is selected. Hence, it's not possible to "really" store the setting in the config file without first selecting another options and then switching back to the first one.

It's not obvious for an admin that the "selected" option from the new setting is not applied.

Reply:
Maybe this is not possible (can the plugin update its own configuration?), but maybe on activate (or somewhere similar?) check the configuration, and update it accordingly?

Maybe I am thinking too much about this :sweat_smile:

Reply:
Sure, a plugin can read and write it's own configuration. But I don't get how that would solve the above UX issue. Given that the first option of the drop down is selected by default, how would admin opt in into the new setting?

Reply:
By default it would be the first option, but if after activate we update the configuration, the option selected will not be the default, but the selected one. Am I right?

Reply:
Yes, you are right, we could write code to handle this case. But I don't think it's worth the complexity. I would much rather just announce this change in the release notes and move on cleanly.